From eb75c066e614154bcabfea023f55c4419ab27f91 Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Tue, 6 Oct 2020 09:38:07 -0700
Subject: [PATCH 1707/1852] staging: xlnxsync: Switch to rising edge interrupt

commit 96b772fbc6d05ae5f4d99a386a7499ab5bd673fc from
https://github.com/Xilinx/linux-xlnx.git

Updated interrupt to rising edge as new SyncIP RTL
generates edge triggered interrupts.

Signed-off-by: Devarsh Thakkar <devarsh.thakkar@xilinx.com>
Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xlnxsync/xlnxsync.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/xlnxsync/xlnxsync.c b/drivers/staging/xlnxsync/xlnxsync.c
index 992c314ff867..5b0f66e44f40 100644
--- a/drivers/staging/xlnxsync/xlnxsync.c
+++ b/drivers/staging/xlnxsync/xlnxsync.c
@@ -1263,7 +1263,8 @@ static int xlnxsync_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 	ret = devm_request_threaded_irq(xlnxsync->dev, xlnxsync->irq, NULL,
-					xlnxsync_irq_handler, IRQF_ONESHOT,
+					xlnxsync_irq_handler,
+					IRQF_ONESHOT | IRQF_TRIGGER_RISING,
 					dev_name(xlnxsync->dev), xlnxsync);
 
 	if (ret) {
-- 
2.31.1

